core/dbt/cli/main.py

```diff
@@ -289,7 +289,13 @@ def debug(ctx, **kwargs):
 
 
 # dbt deps
-@cli.command("deps")
+@cli.group()
```

Comment:
I figured out why `invoke_without_command=True` doesn't work for this case. Everything in click works, but in Flags we did some not ideal lookup [here](https://github.com/dbt-labs/dbt-core/blob/93f72f4100b7c327532ba0fbd06b3e754de66cf2/core/dbt/cli/flags.py#L105) that made the assumption that a function name should always match the last part of input command. I will create a follow up ticket for that but you will find that if you change function name `deps_install` and `deps_lock` below to just `install` and `lock`, you should be able to run without any issue. One thing that I didn't check is whether the params at group level is correctly represented in Flags.